cucumber/src/main/java/eu/drus/jpa/unit/cucumber/BeanFactory.java

```diff
@@ -0,0 +1,41 @@
+package eu.drus.jpa.unit.cucumber;
+
+import static eu.drus.jpa.unit.cucumber.utils.ClassLoaderUtils.tryLoadClassForName;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Method;
+
+import eu.drus.jpa.unit.api.JpaUnitException;
+
+public final class BeanFactory {
+
+    private BeanFactory() {}
+
+    public static <T> T createBean(final Class<T> clazz) {
+        final Class<?> bpClass = tryLoadClassForName("org.apache.deltaspike.core.api.provider.BeanProvider");
+        if (bpClass != null) {
+            try {
+                final Method getContextualReference = bpClass.getMethod("getContextualReference", Class.class, Annotation[].class);
+                return clazz.cast(getContextualReference.invoke(null, clazz, new Annotation[0]));
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Disambiguate this call by either casting as "Object" or "Object[]". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3878)